components/automate-ui/src/app/modules/desktop/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { map, filter } from 'rxjs/operators';
+import { last, reverse } from 'lodash/fp';
+
+import {
+  GetDailyCheckInTimeSeries,
+  SetDaysAgoSelected
+} from 'app/entities/desktop/desktop.actions';
+
+import {
+  dailyCheckInCountCollection,
+  getSelectedDaysAgo
+} from 'app/entities/desktop/desktop.selectors';
+
+import {
+  DailyCheckInCount, DailyCheckInCountCollection, DayPercentage
+} from 'app/entities/desktop/desktop.model';
+
+@Component({
+  selector: 'app-desktop-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss']
+})
+export class DashboardComponent implements OnInit {
+
+  private checkInCountCollection$: Observable<DailyCheckInCountCollection>;
+  private last24HourCheckInCount$: Observable<DailyCheckInCount>;
+  public unknownPercentage$: Observable<number>;
+  public checkedInPercentage$: Observable<number>;
+  public totalCount$: Observable<number>;
+  public unknownCount$: Observable<number>;
+  public checkedInCount$: Observable<number>;
+  public days$: Observable<DayPercentage[]>;
+  public selectedDaysAgo$: Observable<number>;
+
+  constructor(
+    private store: Store<NgrxStateAtom>
+  ) { }
+
+  ngOnInit() {
+    this.store.dispatch(new GetDailyCheckInTimeSeries());
+
+    this.selectedDaysAgo$ = this.store.select(getSelectedDaysAgo);
+
+    this.checkInCountCollection$ = this.store.select(dailyCheckInCountCollection).pipe(
+      filter(collection => collection.buckets.length > 0));
+
+    this.days$ = this.checkInCountCollection$.pipe(
+      map(collection =>
+        reverse(collection.buckets).map((bucket, index) => {
+          const percentage = (bucket.checkInCount / bucket.total) * 100;
+          return {daysAgo: index, percentage: percentage};
+        }))
+    );
+
+    this.last24HourCheckInCount$ = this.checkInCountCollection$.pipe(
+      map(collection => last(collection.buckets))
+    );
+
+    this.unknownPercentage$ = this.last24HourCheckInCount$.pipe(
+        map(count => ((count.total - count.checkInCount) / count.total) * 100)
+      );
+
+    this.checkedInPercentage$ = this.last24HourCheckInCount$.pipe(
```

Comment:
if both the counts are 0, you'll end up with NaN here, bc 0/0 == NaN

<img width="488" alt="Screen Shot 2020-03-26 at 16 36 06" src="https://user-images.githubusercontent.com/10341541/77703222-ea42d680-6f7f-11ea-89c2-3d3691ba2ea5.png">

Reply:
Oh, that should be %100. I will update it.

Reply:
I pushed a fix for that case.

Reply:
I'm seeing these as %'s now 👍